### ISSUE TITLE
Cannot read property 'clientHeight' of null

### ISSUE THREAD

**Issue by psamim** [DevExpress/devextreme-reactive#1572]
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...
- [x] React Grid
- [ ] React Chart

### Explanation
Everything is working but I am getting this in the browser console:

```
Uncaught TypeError: Cannot read property 'clientHeight' of null
    at Sizer.setupListeners (dx-react-core.es.js?8d60:1609)
setupListeners @ dx-react-core.es.js?8d60:1609
scroll (async)
restoreLastFocus @ Modal.js?a179:276
Modal._this.handleClose @ Modal.js?a179:148
componentWillUnmount @ Modal.js?a179:242
callComponentWillUnmountWithTimer @ react-dom.development.js?61bb:15173
callCallback @ react-dom.development.js?61bb:149
invokeGuardedCallbackDev @ react-dom.development.js?61bb:199
invokeGuardedCallback @ react-dom.development.js?61bb:256
safelyCallComponentWillUnmount @ react-dom.development.js?61bb:15180
commitUnmount @ react-dom.development.js?61bb:15472
commitNestedUnmounts @ react-dom.development.js?61bb:15503
unmountHostComponents @ react-dom.development.js?61bb:15759
commitDeletion @ react-dom.development.js?61bb:15811
commitAllHostEffects @ react-dom.development.js?61bb:16473
callCallback @ react-dom.development.js?61bb:149
invokeGuardedCallbackDev @ react-dom.development.js?61bb:199
invokeGuardedCallback @ react-dom.development.js?61bb:256
commitRoot @ react-dom.development.js?61bb:16642
completeRoot @ react-dom.development.js?61bb:18069
performWorkOnRoot @ react-dom.development.js?61bb:17997
performWork @ react-dom.development.js?61bb:17901
performSyncWork @ react-dom.development.js?61bb:17873
interactiveUpdates$1 @ react-dom.development.js?61bb:18164
interactiveUpdates @ react-dom.development.js?61bb:2217
dispatchInteractiveEvent @ react-dom.development.js?61bb:4916
```

### Environment
- devextreme-reactive: 1.8.0
- react: 16.6
- browser: Google Chrome 70
- bootstrap: none
- react-bootstrap: none
- material-ui: 3.3.2

**Comment by SergeyAlexeev** [comment]
Hi,

Would you please provide more details about your application? What Grid plugins are you using?

**Comment by SergeyAlexeev** [comment]
We are closing the issue due to inactivity. Feel free to reopen it if you need further assistance.

**Comment by MaksimMakarevich** [comment]
I have the same issue. I'm using AutoSizer around a virtual table react grid with remote data loading. When I open page with grid and goes back immediately to prev page I have a error message "Uncaught TypeError: Cannot read property 'clientHeight' of null" in dx-react-core.es.js file. After investigation I found that you add events listening to dom element, but don't remove it after component will unmount.
- devextreme-reactive: 1.10
- react: 16.6.3

**Comment by SergeyAlexeev** [comment]
Hi @MaksimMakarevich ,

Thank you for the additional information, but I still can't reproduce the issue on my side. Would you please modify this [sample](https://codesandbox.io/s/2x3zm7vv1n) to show the problem?

Also, it would be great if you provide more details about navigation between pages:

> When I open page with grid and goes back immediately to prev page I have a error message...

**Comment by MaksimMakarevich** [comment]
Updated [sample](https://codesandbox.io/s/82xxpyyo50)

**Comment by SergeyAlexeev** [comment]
Thank you for your reply.

I've reviewed your sample and noticed that you are using the AutoSizer component in the PluginHost component. As you probably know, the PluginHost component **should contain only plugin primitives and plugin components**. Check the [documentation](https://devexpress.github.io/devextreme-reactive/react/core/docs/guides/fundamentals/) for more information. In other words, it is not possible to use the AutoSizer component directly in 'Grid' tags.

As far as I understand, you want the Grid component to fill all the available space. If so, we already have a solution for this scenario. Check the [demo](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/virtual-scrolling#stretching-out-to-parent-element-size) illustrating this. In addition, take a look at the updated [example](https://codesandbox.io/s/xr7vv50rxz).

**Comment by iposokhin** [comment]
I have the same issue. I'm using ColumnChooser and VirtualTable. When I don't define a rootComponent property on Grid component I have't a error message "Uncaught TypeError: Cannot read property 'clientHeight' of null", but virtual table don't work, but ColumnChooser display dropdown list correctly. When I use rootComponent property on Grid component I have a error message "Uncaught TypeError: Cannot read property 'clientHeight' of null", virtual table work fine, but ColumnChooser display a dropdown list in top left corner.

`<Paper style={{ height: '100%' }}>
        <Grid
          rows={ Object.values( people ) }
          columns={[
            { name: 'id', title: '#' },
            { name: 'title', title: 'Name' },
            { name: 'phone', title: 'Phone' },
            { name: 'email', title: 'Email' }
          ]}
          rootComponent={ Root }
          >
          <SortingState
            sorting={ sorting }
            onSortingChange={ onSortingChange } 
          />
          <IntegratedSorting />
          <VirtualTable height="auto" />
          <TableHeaderRow  showSortingControls sortLabelComponent={ SortLabel }/>
          <TableColumnVisibility
            hiddenColumnNames={ hiddenColumnNames }
            onHiddenColumnNamesChange={ onHiddenColumnNamesChange }
          />
          <Toolbar />
          <Template name="toolbarContent">
            <TemplatePlaceholder />
            <PeopleSettings peopleCount={ Object.values( people ).length }/>
          </Template>
          <ColumnChooser
            messages={ { showColumnChooser: 'show column chooser' } }
          />
        </Grid>
      </Paper>`

Why it happens? 

Please, add more examples on API reference. Just description is not informative. Add a variants of use each property. For example 

> Grid.Root  - a component that renders the grid root layout

 In your examples, `<Grid.Root { ...props } style={{ height: '100%' }} />` everywhere. Can it be different?

**Comment by SergeyAlexeev** [comment]
Hi,

Thank you for the detailed report. 

> When I don't define a rootComponent property on Grid component I have't a error message "Uncaught TypeError: Cannot read property 'clientHeight' of null", but virtual table don't work.

Would you please modify this [example](https://codesandbox.io/s/o983r7plyq) to reproduce the behavior you described?
 
Anyway, we've fixed the initial issue in the context of this [PR](https://github.com/DevExpress/devextreme-reactive/pull/1764).
 
As for our documentation, we are working on its improvement. Thank you for your feedback.

**Comment by MaksimMakarevich** [comment]
Hi! 
Thanks for your help! I found new issue. Please check may be i do something wrong. Open this [example](https://codesandbox.io/s/lrn4wpl7yq).
To reproduce:
- Open grid page
- Resize a browser window to show both scrollbars
- Change window height few times(to show only 3-5 lines in grid)
- Change window height to maximum possible
- Repeat few times

After that manipulations usually I has empty space under virtual table(it's do not fill all places in Grid) and vertical scroll bar. Shall I create a new issue?

**Comment by SergeyAlexeev** [comment]
Hi,

We've already fixed the same issue in #1828 The fix will be available in the next update.
